drivers/counter/counter_mcux_rtc.c

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2018 blik GmbH
+ * Copyright (c) 2018, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <counter.h>
+#include <fsl_rtc.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(mcux_rtc, CONFIG_COUNTER_LOG_LEVEL);
+
+struct mcux_rtc_data {
+	counter_alarm_callback_t alarm_callback;
+	counter_top_callback_t top_callback;
+	void *alarm_user_data;
+	void *top_user_data;
+};
+
+struct mcux_rtc_config {
+	struct counter_config_info info;
+	RTC_Type *base;
+	void (*irq_config_func)(struct device *dev);
+};
+
+static int mcux_rtc_start(struct device *dev)
+{
+	const struct counter_config_info *info = dev->config->config_info;
+	const struct mcux_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_rtc_config, info);
+
+	RTC_StartTimer(config->base);
+	RTC_EnableInterrupts(config->base,
+			     kRTC_AlarmInterruptEnable |
+			     kRTC_TimeOverflowInterruptEnable |
+			     kRTC_TimeInvalidInterruptEnable);
+
+	return 0;
+}
+
+static int mcux_rtc_stop(struct device *dev)
+{
+	const struct counter_config_info *info = dev->config->config_info;
+	const struct mcux_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_rtc_config, info);
+
+	RTC_DisableInterrupts(config->base,
+			      kRTC_AlarmInterruptEnable |
+			      kRTC_TimeOverflowInterruptEnable |
+			      kRTC_TimeInvalidInterruptEnable);
+	RTC_StopTimer(config->base);
+
+	/* clear out any set alarms */
+	config->base->TAR = 0;
+
+	return 0;
+}
+
+static u32_t mcux_rtc_read(struct device *dev)
+{
+	const struct counter_config_info *info = dev->config->config_info;
+	const struct mcux_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_rtc_config, info);
+
+	u32_t ticks = config->base->TSR;
+
+	/*
+	 * Read TSR seconds twice in case it glitches during an update.
+	 * This can happen when a read occurs at the time the register is
+	 * incrementing.
+	 */
+	if (config->base->TSR == ticks) {
+		return ticks;
+	}
+
+	ticks = config->base->TSR;
```

Comment:
it might be possible but i would propose to merge this branch anyway to topic-counters since #11729 is based on that and cannot move forward until this PR goes in. It can be fixed later on. @carlescufi could you merge it? Seems that i don't have rights to do that.

Reply:
Yes, I'm fine with merging this to topic-counters, assuming my concern is addressed in some way (which might be "go away").  I am **not** signing off for merging topic-counters to master at this time.

Reply:
Why 30 us? The counter tick is 1 second

Reply:
I thought I saw something that said the clock ran at 32 KiHz (I see now it's the oscillator).  If it's actually 1 Hz, maybe its rare enough to not be worth doing.